portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,8 +49,16 @@ private String convertToJsonUsingGson(List<String> list) {
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
     String text = getParameter(request, "text-input", "");
+    long timestamp = System.currentTimeMillis();
     comments.add(text);
     response.sendRedirect("/index.html");
+
+    Entity commentEntity = new Entity("Comment");
```

Comment:
You can move this block to a writeToDatastore method
private void writeToDatastore(text, timestamp) {
  Entity commentEntity = ...
  ...
  datastore.put(commentEntity);
}